src/v/archival/ntp_archiver_service.h

```diff
@@ -85,6 +85,29 @@ class ntp_archiver_upload_result {
     cloud_storage::upload_result _result{cloud_storage::upload_result::success};
 };
 
+// Indicates whether the flush() command was accepted or not.
+// In the unaccepted case, the flush() command should be
+// re-dispatched to the partition leader by caller.
+enum class flush_response { rejected, accepted };
+
+std::ostream& operator<<(std::ostream& os, flush_response fr);
+
+struct flush_result {
+    flush_response response;
+    // The offset up to which the archiver will flush(), if response is
```

Comment:
inclusive/non-inclusive?

nit: also seems like optional being engaged/disengaged is enough to encode rejected/accept. but, i'm also for being explicit.

Reply:
Amended comment in order to indicate that `flush_result::offset` is inclusive.

I agree, I wasn't sure if we would have other possible responses in the future and I wanted to be explicit about the response by not relying on `std::optional<>.has_value()`.